ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt

```diff
@@ -68,13 +69,15 @@ interface UtxoTransactionBuilder {
         return addCommandAndSignatories(command, signatories.toSet())
     }
 
-    /**
-     * Adds an input state to the current [UtxoTransactionBuilder].
-     *
-     * @param stateAndRef The [StateAndRef] of the input state to add to the current [UtxoTransactionBuilder].
-     * @return Returns a [UtxoTransactionBuilder] including the additional input state.
-     */
-    fun addInputState(stateAndRef: StateAndRef<*>): UtxoTransactionBuilder
+    fun addInputState(stateRef: StateRef): UtxoTransactionBuilder
+
+//    /**
+//     * Adds an input state to the current [UtxoTransactionBuilder].
+//     *
+//     * @param stateAndRef The [StateAndRef] of the input state to add to the current [UtxoTransactionBuilder].
+//     * @return Returns a [UtxoTransactionBuilder] including the additional input state.
+//     */
+//    fun addInputState(stateAndRef: StateAndRef<*>): UtxoTransactionBuilder
```

Comment:
What is the reasoning behind this change? And why only for inputs, not for reference inputs?